src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -184,6 +185,11 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
         _ <- gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName)
         // Change the cluster status to Deleting in the database
         _ <- dbRef.inTransaction(dataAccess => dataAccess.clusterQuery.markPendingDeletion(cluster.googleId))
+        //Delete the staging bucket
+        _ <- cluster.stagingBucket match {
+          case Some(stagingBucket) => gdDAO.deleteBucket(cluster.googleProject, stagingBucket)
```

Comment:
i'd say leave this one too. (is it weird that there are two codepaths?)

Reply:
Yeah we could probably leave it here too. 

I guess we delete clusters in a lot of places:
1. Explicit API delete: transition from Running -> Deleting
2. Cluster monitor: transition from Deleting -> Deleted
3. Cluster monitor: transition from Creating -> Error.
4. API create, but something goes wrong and we have to clean up

If there's a nice way to deduplicate this stuff I'm all for it! Maybe a method like `cleanUpGoogleResources(Cluster)`?

Just TOL, not something that needs to be done for this PR.